tests/projects/projects_namespace_test.py

```diff
@@ -1,16 +1,201 @@
 
     assert 200 == response.status_code
-
     json_data = json.loads(response.data)
     assert [] == json_data
     repository_find_all_mock.assert_called_once()
+
+
+def test_get_project_should_succeed_with_valid_id(client: FlaskClient, mocker: MockFixture):
+    from time_tracker_api.projects.projects_namespace import project_dao
+
+    valid_id = fake.random_int(1, 9999)
+
+    repository_find_mock = mocker.patch.object(project_dao.repository,
+                                               'find',
+                                               return_value=fake_project)
+
+    response = client.get("/projects/%s" % valid_id, follow_redirects=True)
+
+    assert 200 == response.status_code
+    fake_project == json.loads(response.data)
+    repository_find_mock.assert_called_once_with(str(valid_id))
+
+
+def test_get_project_should_return_not_found_with_invalid_id(client: FlaskClient, mocker: MockFixture):
+    from time_tracker_api.projects.projects_namespace import project_dao
+    from werkzeug.exceptions import NotFound
+
+    invalid_id = fake.random_int(1, 9999)
+
+    repository_find_mock = mocker.patch.object(project_dao.repository,
+                                               'find',
+                                               side_effect=NotFound)
+
+    response = client.get("/projects/%s" % invalid_id, follow_redirects=True)
+
+    assert 404 == response.status_code
+    repository_find_mock.assert_called_once_with(str(invalid_id))
+
+
+def test_get_project_should_return_422_for_invalid_id_format(client: FlaskClient, mocker: MockFixture):
+    from time_tracker_api.projects.projects_namespace import project_dao
+    from werkzeug.exceptions import UnprocessableEntity
+
+    invalid_id = fake.company()
+
+    repository_find_mock = mocker.patch.object(project_dao.repository,
+                                               'find',
+                                               side_effect=UnprocessableEntity)
+
+    response = client.get("/projects/%s" % invalid_id, follow_redirects=True)
+
+    assert 422 == response.status_code
+    repository_find_mock.assert_called_once_with(str(invalid_id))
+
+
+def update_project_should_succeed_with_valid_data(client: FlaskClient, mocker: MockFixture):
+    from time_tracker_api.projects.projects_namespace import project_dao
+
+    repository_update_mock = mocker.patch.object(project_dao.repository,
+                                                 'update',
+                                                 return_value=fake_project)
+
+    valid_id = fake.random_int(1, 9999)
+    response = client.put("/projects/%s" % valid_id, json=valid_project_data, follow_redirects=True)
+
+    assert 200 == response.status_code
+    fake_project == json.loads(response.data)
+    repository_update_mock.assert_called_once_with(valid_id, valid_project_data)
+
+
+def test_update_project_should_reject_bad_request(client: FlaskClient, mocker: MockFixture):
```

Comment:
I think we can use the name `test_update_project_should_fail_with_invalid_request` as before.

Reply:
Bad request is better